common/src/main/java/com/viaversion/viaversion/protocol/ProtocolManagerImpl.java

```diff
@@ -365,11 +371,20 @@ > VersionedPacketTransformer<C, S> createPacketTransformer(ProtocolVersion input
     }
 
     @Override
-    public List<Protocol> getBaseProtocols(ProtocolVersion serverVersion) {
+    public List<Protocol> getBaseProtocols(@Nullable ProtocolVersion clientVersion, @Nullable ProtocolVersion serverVersion) {
```

Comment:
Any reason for them being nullable?

Reply:
At least the `clientboundBaseProtocolVersion` in InitialBaseProtocol will be null if no valid base protocol can be found

Reply:
Is that actually a valid client state?

Reply:
Yes, special versions not using the RedirectProtocolVersion can't have the base protocol set via the registry (as they might compare equal to normal versions). In this case, they always need to manually add the base protocol.

Reply:
> clientboundBaseProtocolVersion

I don't see any referceny that says its a state. Others like handshake, login, ping etc are

Reply:
This is not related to protocol states but just to whether it's legitimate to not have a clientbound base protocol found in the registry.